src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -61,19 +61,33 @@ def auth(self, url: str, cluster_alias: str, username: str,
                 'username': username,
                 'password': password
             }
-            content = self._proxy('POST', url, 'api/auth', payload=payload)
-            if 'token' not in content:
-                raise DashboardException(
-                    "Could not authenticate to remote cluster",
-                    http_status_code=400,
-                    component='dashboard')
+            try:
+                content = self._proxy('POST', url, 'api/auth', payload=payload,
+                                      verify=ssl_verify, cert=ssl_certificate)
+                if 'token' not in content:
+                    return content['detail']
+                user_content = self._proxy('GET', url, f'api/user/{username}',
+                                           token=content['token'])
+                if 'status' in user_content and user_content['status'] == '403 Forbidden':
+                    return 'User is not an administrator'
```

Comment:
we cannot assume a user as an administrator if the status is not Forbidden. If you check the role `cluster-manager` he has the previlleage to read/write config-opt and lacks permissions in may other components. So this won't hold true everytime. And moreover we should raise exceptions when we are excepting failures like in the case of denied access. And to properly check if the role is of admin, we should handle the initial exceptions and then we should go for the actual check if you can find an administrator role in the given user.

Reply:
Updated, Thanks